cmd/up/main.go

```diff
@@ -136,8 +136,8 @@ func main() {
 			level.Info(l).Log("msg", "start querying", "type", opts.EndpointType)
 
 			return runPeriodically(ctx, opts, m.QueryResponses, l, ch, func(rCtx context.Context) {
-				if err := read(rCtx, l, m, opts); err != nil {
-					m.QueryResponses.WithLabelValues(labelError).Inc()
+				if httpCode, err := read(rCtx, l, m, opts); err != nil {
+					m.QueryResponses.WithLabelValues(labelError, getHttpCode(httpCode)).Inc()
```

Comment:
Honestly - I don't think it is necessary to include the `getHttpCode` function at all - looking at the `http_requests_total` [metric](https://prometheus.telemeter-prod-01.devshift.net/graph?g0.range_input=1h&g0.expr=http_requests_total&g0.tab=1) for inspiration, they observe the error code directly which would be my preference too.

Reply:
`getHttpCode` is in place to avoid shooting up cardinality. If we can think that cardinality is not an issue, we can capture the raw HTTP code.

Reply:
removed `gettHttpCode` , we can revert to this function if the cardinality becomes an issue to worry about.